modules/apps/object/object-test/src/testFunctional/tests/ObjectNotifications.testcase

```diff
@@ -749,6 +749,125 @@ definition {
 		}
 	}
 
+	@description = "LPS-165849 - Verify terms from a related custom object (child) are replaced correctly in a user notification"
+	@priority = 5
+	test CanSendUserNotificationWithTermsComingFromCustomObjectRelationship {
+		property custom.properties = "feature.flag.LPS-165849=true";
+		property portal.acceptance = "true";
+
+		task ("Given two custom objects, A and B, each with a text field and publish them") {
+			for (var letters : list "A,B") {
+				ObjectAdmin.addObjectViaAPI(
+					labelName = "Custom Object ${letters} 182607",
+					objectName = "CustomObject${letters}182607",
+					pluralLabelName = "Custom Objects ${letters} 182607");
+
+				ObjectAdmin.addObjectFieldViaAPI(
+					fieldBusinessType = "Text",
+					fieldLabelName = "Field Text ${letters}",
+					fieldName = "customField",
+					fieldType = "String",
+					isRequired = "false",
+					objectName = "CustomObject${letters}182607");
+
+				ObjectAdmin.publishObjectViaAPI(objectName = "CustomObject${letters}182607");
+			}
+		}
+
+		task ("And given a 1:M relationship between the Object A and Object B") {
+			ObjectAdmin.addObjectRelationshipViaAPI(
+				objectName_1 = "CustomObjectA182607",
+				objectName_2 = "CustomObjectB182607",
+				relationshipLabel = "Relationship",
+				relationshipName = "relationship",
+				relationshipType = "oneToMany");
+		}
+
+		task ("And given a User Custom with admin roles is added") {
+			JSONUser.addUser(
+				userEmailAddress = "userea@liferay.com",
+				userFirstName = "userfn",
+				userLastName = "userln",
+				userScreenName = "usersn");
+
+			JSONRole.assignRoleToUser(
+				roleTitle = "Administrator",
+				userEmailAddress = "userea@liferay.com");
+		}
+
+		task ("And given a user notification template is created using terms from both object definitions") {
```

Comment:
"And given a user notification template is created using terms from both objects"